packages/resolve-redux/src/connect_read_model.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import actions from './actions'
+
+export default (
+  mapStateToProps,
+  mapDispatchToProps,
+  mergeProps,
+  options
+) => Component => {
+  const ConnectedComponent = connect(
+    mapStateToProps,
+    mapDispatchToProps,
+    mergeProps,
+    options
+  )(Component)
+
+  class ReadModelConnector extends React.PureComponent {
+    componentWillReceiveProps(nextProps) {
+      if (
+        JSON.stringify(nextProps.variables) ===
+          JSON.stringify(this.variables) &&
+        nextProps.readModelName === this.readModelName &&
+        nextProps.resolverName === this.resolverName &&
+        nextProps.isReactive === this.isReactive &&
+        nextProps.query === this.query
```

Comment:
Smell (imho)

Reply:
Lifecycle for `componentWillReceiveProps` has an requirement to perform manual comparation between old & new props and actual state, so such kind comparation check is mandatory to avoid infinite loop.